### PR TITLE
De(structable/constructible) objects rework

### PR DESCRIPTION
**Pull request by PhantomEpicness**
## Abstract
Dozens of snowflake and legacy objects are straight up indestructible in CM through normal means. Most people who play CM regulary grown used to it but it is 
shockingly apparent when you actually to try to notice it or when someone abuses it. Things like filing cabinets, hydroponics trays ore boxes, storage tanks, showcases 
and computers are indestructible through normal means (if at all) for god-knows-why and are constantly used for cheesing every now and then. If the CLF threw away their armor and
just wore crates (100% bulletproof), they would control the galaxy already.

Furthermore, many things also lack proper deconstruction. You should be able to deconstruct nearly everything destructible ( within balance bounds, of course)

***Disclaimer: UNINTENTIONAL objects meaning things that aren't meant to be indestructable or affect balance is a big way if its destructable, like hulls ***

## Goals

1. Find nearly all ***unintentional*** indestructable things into being destructable and deconstructable
3. Give nearly all of the above deconstruction paths.
  - Deconstructable objects should give a reasonable amount of mats depending on the rarity of the obj and how complex the deconstruction path is. 
  Exotic machinery found in atmos or the handful of generators should gain exotic materials logically.
  - ex: Big Scrubbers yielding plasteel after some time of deconstructing
 
 Deconning objects should ALWAYS give more rewards than simply destroying

3. Give some of the more used objects actual recipes to build them back again
  - If you are able to deconstruct it, you should be able to build it again
  - Things like hydroponics trays that are commonly used so entire roles wont get screwed because it got deconstructed.
  - Not everything should have it, of course. There's already enough snowflake objects, we dont need more.

4. Give crafting recipes to stuff that SHOULD have it but do not
  - Things like wooden tables, pool tables, carpets,  etc. Why on earth isnt there an easy way to make them. MT buff.

## Non-goals

1. Nothing should be affecting balance in an adverse way
  - We obviously don't want people getting entire stacks of plasteel from 1 area.
  - This is not a major balance affecting MR and shouldn't be.

2. Intentionally indestructable objects should not be touched
  - this has been repeated throughout this doc but putting it here so someone won't say *"SUPER MARINE/XENO BUFF THEY CAN DESTROY HULL WALLS!!!111"*

## Content

1. Rework nearly all unintentional indestructable things into being destructable and deconstructable

  Quite self explanitory, every month a new cheese obj comes out that has to be banned from APC/Dropship or made unmovable/slashable for xenos. Putting our foot down
  and focusing on ending it will greatly improve QoL with only minimal impact on balance and a whole lot of sanity saved in making bandaid fixes.
  
  ***EVERYTHING that is unintentionally indesdtructable NEEDS to be destructable***
  
  Computers & vendors will be an issue since some of them SHOULD not be deconnable/destructable like intel computers, but thats something that could easily be fixed with a subtype/var

  Things like crates which contain rewards inside should be destructable with bullets/melee, but have a 50% chance of destroying contents

  This could be done by giving everything health vars and damage procs like seen in showcases or barricades. 
  
  Some things that were given snowflake anchoring/indestructable could be taken a look at again and be made deconstructable again.
  
  There are potential balance ramifications coming from this due to being more freedom of movement in the battlefield, but I believe freedom of movement is good.
  *A 15 foot rav shouldn't die because a dinky little filing cabinet blocked their way*

2. Give destructable objects deconstruction paths.

  This is quite a bit more complex. We want to give them unique good decontructable paths that aren't too hard and gives an amount of mats equal to the rarity of the object
  in question. More rarer and exotic objects like big scrubbers should obviously be harder to deconstruct, but rewarding more valuable mats in return.
  
  The primary goal here is to have the majority of *unintentionally* destructable objects be deconstructable.
  
  Giving more opportunities for engies to loot colony side will theorectically promote them salvage more and allow for them to have a little bit more 
  independence from req by trading some element of risk. Potential RP opportunities are a plus side.

  *ex: FOB runs out of mats and REQ isn't responding. Bravo SL takes the initiative and leads a salavaging team to colony atmos to deconstruct some machinery.*
  
3. Give some of the more used objects actual recipes to build them back again

  More or less self explanitory. We dont want people's experience shafted because a cheeky MT deconstructed their fridge or hydroponics tray. Used objects should be
  able to be built back at a reasonable cost.
  
4. Give crafting recipes to stuff that SHOULD have it but do not

   There's still a few dumb things that should need crafting recipes like wooden tables and stuff like that. Stuff thats borderline unobtainable for 0 reason at all 
   should be made deconstructable. Decor objects should obviously have recipes for RP.
   
   
With all goals done it should give CM:
  - Limited potential balance ramifications
     - More freedom of movement would cause this.
     - Mats would be more avalible colonyside. This wouldn't be too large since you trade risk for reward (lurkers) but its still a thing.
  - Massive QoL for xenos and marines alike.
    - Trust me, it may not seem like much, but this would change how a lot of us play because we instictually avoid snowflake objs.
  - More engineer salvaging opporunities and req independence.
  - Some more RP opporunities generated from decor objects being accessible.

   

## Potential Changes

- Reward bias being given toward objs found colonyside
  - This could be done to promote more risky,but rewarding salvaging colonyside rather than COs bitching at MTs to deconstruct the entire ship every round.

### CODE REVIEW

File: deconnable_everything.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+## Abstract
+Dozens of snowflake and legacy objects are straight up indestructible in CM through normal means. Most people who play CM regulary grown used to it but it is 
+shockingly apparent when you actually to try to notice it or when someone abuses it. Things like filing cabinets, hydroponics trays ore boxes, storage tanks, showcases 
+and computers are indestructible through normal means (if at all) for god-knows-why and are constantly used for cheesing every now and then. If the CLF threw away their armor and
+just wore crates (100% bulletproof), they would control the galaxy already.
+
+Furthermore, many things also lack proper deconstruction. You should be able to deconstruct nearly everything destructible ( within balance bounds, of course)
+
+***Disclaimer: UNINTENTIONAL objects meaning things that aren't meant to be indestructable or affect balance is a big way if its destructable, like hulls ***
+<!-- An abstract is a short blurb, about a paragraph or two, succinctly describing your feature. This should mostly be "why", but can include "what". -->
```
Comment (Watermelon914) [inline deconnable_everything.md:12]: This should be a description of what most objects should be like, aka destructible. Doesn't need to be long or anything, just justify why they should be destructible. Don't refer to how things were in the past

Comment (Watermelon914) [inline deconnable_everything.md:12]: This should be a description of what most objects should be like, aka destructible. Doesn't need to be long or anything, just justify why they should be destructible. Don't refer to how things were in the past

File: deconnable_everything.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+ Deconning objects should ALWAYS give more rewards than simply destroying
+
+3. Give some of the more used objects actual recipes to build them back again
+  - If you are able to deconstruct it, you should be able to build it again
+  - Things like hydroponics trays that are commonly used so entire roles wont get screwed because it got deconstructed.
+  - Not everything should have it, of course. There's already enough snowflake objects, we dont need more.
+
+4. Give crafting recipes to stuff that SHOULD have it but do not
+  - Things like wooden tables, pool tables, carpets,  etc. Why on earth isnt there an easy way to make them. MT buff.
+<!-- This is a numbered list clearly detailing your goals for the feature. As per usual, this should be a mixture of both why and what. -->
```
Comment (Watermelon914) [inline deconnable_everything.md:31]: This is all implementation detail. This does not fit in a design document.

Comment (Watermelon914) [inline deconnable_everything.md:31]: This is all implementation detail. This does not fit in a design document.

File: deconnable_everything.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+## Non-goals
+
+1. Nothing should be affecting balance in an adverse way
+  - We obviously don't want people getting entire stacks of plasteel from 1 area.
+  - This is not a major balance affecting MR and shouldn't be.
+
+2. Intentionally indestructable objects should not be touched
+  - this has been repeated throughout this doc but putting it here so someone won't say *"SUPER MARINE/XENO BUFF THEY CAN DESTROY HULL WALLS!!!111"*
+<!-- Just like goals, but the opposite! Every feature has boundaries it won't step over. These should be written as if they start with "We will not...". -->
```
Comment (Watermelon914) [inline deconnable_everything.md:41]: Again, this is including implementation detail, it should only include design detail.

Comment (Watermelon914) [inline deconnable_everything.md:41]: Again, this is including implementation detail, it should only include design detail.

File: deconnable_everything.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  - More engineer salvaging opporunities and req independence.
+  - Some more RP opporunities generated from decor objects being accessible.
+
+   
+
+<!-- Now's where you get into clear detail about everything your feature does. **You should still be explaining 'why' things are that way, *as* you describe what.** Be as detailed as possible. -->
+
+## Alternatives
+
+<!-- Provide potential alternatives to your feature, either ones that align with your design values, or ones that don't that you suspect will be suggested. If you are including the latter, make sure to explain why you didn't choose that. -->
```
Comment (Watermelon914) [inline deconnable_everything.md:101]: Get rid of this section if you aren't using it

Comment (Watermelon914) [inline deconnable_everything.md:101]: Get rid of this section if you aren't using it

File: deconnable_everything.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  Things like crates which contain rewards inside should be destructable with bullets/melee, but have a 50% chance of destroying contents
+
+  This could be done by giving everything health vars and damage procs like seen in showcases or barricades. 
+  
+  Some things that were given snowflake anchoring/indestructable could be taken a look at again and be made deconstructable again.
+  
+  There are potential balance ramifications coming from this due to being more freedom of movement in the battlefield, but I believe freedom of movement is good.
+  *A 15 foot rav shouldn't die because a dinky little filing cabinet blocked their way*
+
+2. Give destructable objects deconstruction paths.
```
Comment (Watermelon914) [inline deconnable_everything.md:63]: Implementation detail. Don't include code-like words in design documents, a designer does not care about deconstruction paths. Reword this into 'Give destructible objects a thing they break down into'

Comment (Watermelon914) [inline deconnable_everything.md:63]: Implementation detail. Don't include code-like words in design documents, a designer does not care about deconstruction paths. Reword this into 'Give destructible objects a thing they break down into'

File: deconnable_everything.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  - We obviously don't want people getting entire stacks of plasteel from 1 area.
+  - This is not a major balance affecting MR and shouldn't be.
+
+2. Intentionally indestructable objects should not be touched
+  - this has been repeated throughout this doc but putting it here so someone won't say *"SUPER MARINE/XENO BUFF THEY CAN DESTROY HULL WALLS!!!111"*
+<!-- Just like goals, but the opposite! Every feature has boundaries it won't step over. These should be written as if they start with "We will not...". -->
+
+## Content
+
+1. Rework nearly all unintentional indestructable things into being destructable and deconstructable
```
Comment (Watermelon914) [inline deconnable_everything.md:45]: This is referring to how things are currently. No designer or future dev is going to care about how things were, they only need to care about how things are right now

Comment (Watermelon914) [inline deconnable_everything.md:45]: This is referring to how things are currently. No designer or future dev is going to care about how things were, they only need to care about how things are right now

File: deconnable_everything.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+## Alternatives
+
+<!-- Provide potential alternatives to your feature, either ones that align with your design values, or ones that don't that you suspect will be suggested. If you are including the latter, make sure to explain why you didn't choose that. -->
+
+## Potential Changes
+
+- Reward bias being given toward objs found colonyside
+  - This could be done to promote more risky,but rewarding salvaging colonyside rather than COs bitching at MTs to deconstruct the entire ship every round.
+<!-- Most of the time you're not going to get the best design first try. It helps to try your best to predict what *could* go wrong, and suggest alternatives that can be taken, without sacrificing your design. -->
```
Comment (Watermelon914) [inline deconnable_everything.md:107]: Not what the potential changes section is for. Remove this section if you don't have any potential changes that may need to be done.
This section is about predicting what could go wrong with your feature, e.g. xenos deconstructing everything, and to suggest alternatives that still align with your design.

Comment (Watermelon914) [inline deconnable_everything.md:107]: Not what the potential changes section is for. Remove this section if you don't have any potential changes that may need to be done.
This section is about predicting what could go wrong with your feature, e.g. xenos deconstructing everything, and to suggest alternatives that still align with your design.

**Review by Nanu308 — COMMENTED** [review]
The essence of this Design Doc itself is fine, however resolve Walter's comments.

**Pull request closed without merging**